addons/mail/models/mail_message.py

```diff
@@ -710,6 +710,36 @@ def set_message_done(self):
             }
         )
 
+    def mark_unread(self):
+        """
+        Marks the current message as unread.
+        """
+        self.ensure_one()
+        if self.model != 'mail.channel':
```

Comment:
Move to channel model. Makes no sense to have a method intended for channels on message model, it should be model agnostic.